### PR TITLE
docs: Add rfc for compact table via sql

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref #4897

Problem Summary:

### What is changed and how it works?

Add a RFC for the compact table via SQL feature.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by breezewish** [comment]
@lidezhu @JaySon-Huang @flowbehappy Hi, this RFC is not urgent. Please take a look when you are free. Thanks!

**Review by lidezhu — APPROVED** [review]
LGTM

File: docs/design/2022-05-19-compact-table-via-sql.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+### Interrupt Execution (Kill)
+
+![](./images/2022-05-19-compact-table-via-sql-2.png)
+
+When user executes `KILL [TIDB] <ID>` to interrupt the compaction command,
+TiDB simply stops the execution by not sending future `CompactRequest`.
+There will be `CompactRequest` running in TiFlash instances at the moment when
+user initiates the `KILL`. These running "small" compaction tasks will be
+untouched and kept running until finished. There is no way to stop a running
```
Comment (leiysky) [inline docs/design/2022-05-19-compact-table-via-sql.md:120]: What will happen to the intermediate files if a compact task is stopped?

Comment (breezewish) [reply docs/design/2022-05-19-compact-table-via-sql.md:120]: TiFlash only accepts segment level compaction requests and these requests cannot be stopped once started. This means, the compaction for one segment will be finished. TiDB will not send more segment level compaction requests, causing the resource usage to drop after KILLing the TiDB SQL execution.

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**